### PR TITLE
Warn when using cloud sdk credentials

### PR DESCRIPTION
**Pull request by TheRoyalTnetennba**
A ruby version of [the google-auth-library-python pr](https://github.com/GoogleCloudPlatform/google-auth-library-python/pull/266) in response to [this issue](https://github.com/google/google-auth-library-ruby/issues/143).

### CODE REVIEW

File: lib/googleauth/client_id.rb [hunk 1]
```diff
@@ -63,13 +64,14 @@ class << self
       #       & secrets in source. See {#from_file} to load from
       #       `client_secrets.json` files.
       def initialize(id, secret)
+        CredentialsLoader.warn_if_cloud_sdk_credentials(id)
```
Comment (dazuma) [inline lib/googleauth/client_id.rb]: We haven't been style-consistent with this repo in the past, but moving forward I'd like to be more so (so eventually we can set the rubocop file to be similar to the one in google-cloud-ruby). Our ruby style is "seattle style" which, among other things, means minimal use of parens. Here, and in the _new_ code you've added below, please drop parens for most normal method calls. e.g.

```ruby
        CredentialsLoader.warn_if_cloud_sdk_credentials id
```

File: lib/googleauth/client_id.rb [hunk 2]
```diff
@@ -63,13 +64,14 @@ class << self
...
         raise 'Client id can not be nil' if id.nil?
         raise 'Client secret can not be nil' if secret.nil?
         @id = id
         @secret = secret
       end
 
-      # Constructs a Client ID from a JSON file downloaed from the
+      # Constructs a Client ID from a JSON file downloaded from the
```
Comment (dazuma) [inline lib/googleauth/client_id.rb]: Thanks for fixing these typos. 👍

File: lib/googleauth/client_id.rb (hunk 1 above)
Comment (TheRoyalTnetennba) [reply lib/googleauth/client_id.rb]: Is there a definitive reference for this I can refer to?

Comment (dazuma) [reply lib/googleauth/client_id.rb]: I don't think so. Best I can point you to is the `.rubocop.yml` files in google-cloud-ruby. ([Example](https://github.com/GoogleCloudPlatform/google-cloud-ruby/blob/master/google-cloud-storage/.rubocop.yml)). The existing code isn't yet compliant with those guidelines, but we should start moving in that direction. Ideally, all our libraries' rubocop settings would be essentially identical.

**Pull request merged**